src/components/Table/GridTable.test.tsx

```diff
@@ -710,6 +710,62 @@ describe("GridTable", () => {
         ),
       ).rejects.toThrow("Column 0 passed an unsortable value, use GridCellContent or clientSideSort=false");
     });
+
+    it("can pin rows and filter - filterable pin", async () => {
+      // Given the table is using client-side sorting
+      const r = await render(
+        <GridTable<Row>
+          filter={"e"}
+          columns={[nameColumn, valueColumn]}
+          sorting={{ on: "client", initial: [1, "ASC"] }}
+          rows={[
+            simpleHeader,
+            // And the 1st row is pinned first as a filterable pin
+            { kind: "data", id: "1", pin: { at: "first", filter: true }, data: { name: "a", value: 11 } },
+            { kind: "data", id: "2", pin: { at: "first", filter: true }, data: { name: "b", value: 10 } },
+            // And the middle rows need sorted
+            { kind: "data", id: "3", data: { name: "c", value: 3 } },
+            { kind: "data", id: "4", data: { name: "d", value: 1 } },
+            // And the last rows are pinned last
+            { kind: "data", id: "5", pin: { at: "last", filter: true }, data: { name: "e", value: 20 } },
+            { kind: "data", id: "6", pin: { at: "last", filter: true }, data: { name: "f", value: 21 } },
+          ]}
+        />,
+      );
+      // Then the e row stayed first and others filtered
+      expect(cell(r, 1, 0)).toHaveTextContent("e");
+    });
+
+    it("can pin rows - filterable pin", async () => {
+      // Given the table is using client-side sorting
+      const r = await render(
+        <GridTable<Row>
+          columns={[nameColumn, valueColumn]}
+          sorting={{ on: "client", initial: [1, "ASC"] }}
+          rows={[
+            simpleHeader,
+            // And the 1st row is pinned first as a filterable pin
+            { kind: "data", id: "1", pin: { at: "first", filter: true }, data: { name: "a", value: 11 } },
+            { kind: "data", id: "2", pin: { at: "first", filter: true }, data: { name: "b", value: 10 } },
+            // And the middle rows need sorted
+            { kind: "data", id: "3", data: { name: "c", value: 3 } },
+            { kind: "data", id: "4", data: { name: "d", value: 1 } },
+            // And the last rows are pinned last
+            { kind: "data", id: "5", pin: { at: "last", filter: true }, data: { name: "e", value: 20 } },
+            { kind: "data", id: "6", pin: { at: "last", filter: true }, data: { name: "f", value: 21 } },
+          ]}
+        />,
+      );
+      // Then the a/b rows stayed first
```

Comment:
Suggestion: Similar to the story, I feel we copy/pasted an already-complicated / dedicated-boundary-case test...like when I read these assertions we're talking about first/middle/last rows changing orders...but none of that is really related to the change in code you just made.

I'd be tempted to remove this 2nd test, and make the 1st test as simple as possible by only having 3 rows, one with `filter: true` (that does not match the filter so is not shown) one with `filter: false` (that does not match the filter but _is_ shown), and one w/o any pin/filter set (that does not match so is not shown).